providers/meteoblue.go

```diff
@@ -45,16 +46,26 @@ func NewMeteoBlueProvider() (*MeteoBlueProvider, error) {
 		client.EnableTrace()
 	}
 
-	return &MeteoBlueProvider{
+	provider := MeteoBlueProvider{
 		client: client,
 		config: meteoConfig,
-	}, nil
+	}
+	// Setting the default location to 0,0
+	provider.SetQueryParams(plumber.NewCoordinates(0, 0))
+
+	// Creating the new request(which will be reused in all FetchData calls) and setting the default queryParams on the client
+	provider.client.NewRequest()
+	provider.client.SetQueryParams(provider.queryParams)
+	return &provider, nil
 }
 
-func (p *MeteoBlueProvider) FetchData(qp map[string]string) (*plumber.BaseData, error) {
-	// Creating a new request everytime we fecth data
-	p.client.NewRequest()
-	resp, err := p.client.SetQueryParams(qp).Get(p.config.APIPath)
+func (p *MeteoBlueProvider) FetchData(coords *plumber.Coordinates) (*plumber.BaseData, error) {
+	resp, err := p.client.
+		SetQueryParams(map[string]string{
+			"latitude":  fmt.Sprintf("%f", coords.Latitude),
+			"longitude": fmt.Sprintf("%f", coords.Longitude),
```

Comment:
Open-Meteo uses "latitude" and "longitude" while Meteoblue uses "lat" and "lon". 

This forms the URL as `"...&lat=0.000000&latitude=10.901838&lon=0.000000&longitude=76.899845&tz=GMT"`

Reply:
Hmm, then how about we don't set the latitude and longitude / lat - long while setting the default query params and directly set them for the first time in the FetchData() call?

Reply:
I reckon that should be okay.

Reply:
Done